ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessService.scala

```diff
@@ -210,15 +214,21 @@ class DBProcessService(managerActor: ActorRef,
       val emptyCanonicalProcess = newProcessPreparer.prepareEmptyProcess(command.processName.value, processingType, command.isSubprocess)
       val action = CreateProcessAction(command.processName, command.category, emptyCanonicalProcess, processingType, command.isSubprocess)
 
-      repositoryManager
-        .runInTransaction(processRepository.saveNewProcess(action))
-        .map{
-          case Right(maybemaybeCreated) =>
-            maybemaybeCreated
-              .map(created => Right(toProcessResponse(command.processName, created)))
-              .getOrElse(Left(ProcessValidationError("Unknown error on creating scenario.")))
-          case Left(value) =>
-            Left(value)
+      val errorsWithoutEmptyProcess = validateInitialScenario(emptyCanonicalProcess, processingType, command.category)
+
+      if (errorsWithoutEmptyProcess.nonEmpty) {
+        Future(Left(ProcessValidationError(errorsWithoutEmptyProcess.head.message)))
```

Comment:
Future.apply is asynchronous. You can use Future.successfull to avoid changing of thread context